### ISSUE TITLE
The BSON serializer WriteCString() bug with wide unicode chars

### ISSUE THREAD

**Issue by itadapter** [aumcode/nfx#57]
```
 [System.ArgumentException] The output byte buffer is too small to contain the encoded data, encoding 'Unicode (UTF-8)' fallback 'System.Text.EncoderReplacementFallback'.
```
from BSON.BinUtils.WriteCSTring(stream, value)

**Comment by itadapter** [comment]
Created NFX.NUnit.Serialization.WriteReadLongStringOfLongChars() - 
it fails with chinese '久'

**Comment by itadapter** [comment]
Fixed by  829b4a8
